### DISCUSSION TITLE
Merge back-end based on merge-ort

### DISCUSSION THREAD

**Discussion by arxanas** [arxanas/git-branchless#77] [Ideas]
Most likely, in-memory rebases will have to duplicate merging logic, since both Git/Libgit2 don't seem suited to carry out in-memory rebases in a performant way. (Currently, Git doesn't support it, and Libgit2 is quite slow if the index is large.)

There's a good overview on the merge-ort strategy here, with a link to a very detailed series of blog posts discussing various optimizations: https://github.blog/2021-08-16-highlights-from-git-2-33/#merge-ort-a-new-merge-strategy

@newren: thank you very much for taking the time to write up those posts!

**Comment by newren** [comment]
Glad others found it interesting/useful.  :-)
